File: chatwoot/java/src/main/java/oseg/chatwoot_examples/CreateAnIntegrationHookExample.java
package oseg.chatwoot_examples;

import com.chatwoot.client.ApiException;
import com.chatwoot.client.Configuration;
import com.chatwoot.client.api.*;
import com.chatwoot.client.auth.*;
import com.chatwoot.client.JSON;
import com.chatwoot.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class CreateAnIntegrationHookExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        config.setApiKey("USER_API_KEY");
        // config.setApiKey("AGENT_BOT_API_KEY");
        // config.setApiKey("PLATFORM_APP_API_KEY");

        var integrationsHookCreatePayload = new IntegrationsHookCreatePayload();
        integrationsHookCreatePayload.appId(null);
        integrationsHookCreatePayload.inboxId(null);

        try
        {
            var response = new IntegrationsApi(config).createAnIntegrationHook(
                null, // accountId
                integrationsHookCreatePayload // data
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling IntegrationsApi#createAnIntegrationHook");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: chatwoot/kotlin/src/main/kotlin/oseg/chatwoot_examples/CreateAnIntegrationHookExample.kt
package oseg.chatwoot_examples

import com.chatwoot.client.infrastructure.*
import com.chatwoot.client.apis.*
import com.chatwoot.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class CreateAnIntegrationHookExample
{
    fun createAnIntegrationHook()
    {
        ApiClient.apiKey["api_access_token"] = "USER_API_KEY"
        // ApiClient.apiKey["api_access_token"] = "AGENT_BOT_API_KEY"
        // ApiClient.apiKey["api_access_token"] = "PLATFORM_APP_API_KEY"

        val integrationsHookCreatePayload = IntegrationsHookCreatePayload()

        try
        {
            val response = IntegrationsApi().createAnIntegrationHook(
                accountId = 0,
                _data = integrationsHookCreatePayload,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling IntegrationsApi#createAnIntegrationHook")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling IntegrationsApi#createAnIntegrationHook")
            e.printStackTrace()
        }
    }
}
